middleware.go

```diff
@@ -0,0 +1,107 @@
+package jwt
+
+import (
+	"errors"
+	"github.com/Digital-Insight-Technologies-Ltd/gin-jwt-cognito/models"
+	"github.com/sirupsen/logrus"
+	"net/http"
+	"strings"
+
+	"github.com/gin-gonic/gin"
+	"github.com/golang-jwt/jwt/v5"
+	"github.com/google/uuid"
+)
+
+// UserClaimsKey is the key for user claims in our context
+const UserClaimsKey = "userClaims"
+
+// CognitoClaimsMiddleware is a middleware for Gin that extracts user claims from
+// AWS Cognito.
+func CognitoClaimsMiddleware(jwtParser *jwt.Parser) gin.HandlerFunc {
+	return func(c *gin.Context) {
+		authHeader := c.GetHeader("Authorization")
+		if !strings.HasPrefix(authHeader, "Bearer ") {
+			c.AbortWithStatusJSON(http.StatusUnauthorized, gin.H{"error": "Missing or invalid Authorization header"})
+			return
+		}
+		claims, err := parseClaims(jwtParser, authHeader[7:])
+
+		if err != nil {
+			logrus.WithError(err).Error("Error parsing claims (OIDC: stage 1)")
+			c.AbortWithStatusJSON(http.StatusUnauthorized, gin.H{"error": "Invalid user claims (OIDC: stage 1)"})
```

Comment:
Enhance error messages to avoid exposing sensitive information. For instance, use generic error messages instead of specifying "stage 1" or "stage 2". [medium]